src/views/bridge-confirmation/components/approval-info/approval-info.view.tsx

```diff
@@ -0,0 +1,37 @@
+import { FC } from "react";
+import { constants as ethersConstants } from "ethers";
+
+import { Token } from "src/domain";
+import { AsyncTask } from "src/utils/types";
+import { ReactComponent as InfoIcon } from "src/assets/icons/info.svg";
+import useApprovalInfoStyles from "src/views/bridge-confirmation/components/approval-info/approval-info.styles";
+import Typography from "src/views/shared/typography/typography.view";
+
+interface ApprovalInfoProps {
+  token: Token;
+  hasAllowanceTask: AsyncTask<boolean, string>;
```

Comment:
I don't like this too much... I think this component should not be responsible for deciding whether it should be rendered, but its parent.

I'd prefer the cases where this component returns `null` not to exist (`token.address === ethersConstants.AddressZero` and `hasAllowanceTask.status !== "successful" || (hasAllowanceTask.status === "successful" && hasAllowanceTask.data)`).

Reply:
Done!